src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexGroup.cs

```diff
@@ -15,19 +15,21 @@ namespace System.Text.RegularExpressions
     public class Group : Capture
     {
         // the empty group object
-        internal static Group _emptygroup = new Group(String.Empty, Array.Empty<int>(), 0);
+        internal static Group _emptygroup = new Group(String.Empty, Array.Empty<int>(), 0, string.Empty);
 
         internal int[] _caps;
         internal int _capcount;
         internal CaptureCollection _capcoll;
+        internal string _name;
```

Comment:
Yes, will make _name and _caps readonly. _capcount is set in Match, which derives from Group.